server/models/Collective.js

```diff
@@ -1472,12 +1472,14 @@ export default function (Sequelize, DataTypes) {
 
     const urlPath = await this.getUrlPath();
     const memberCollective = await models.Collective.findByPk(member.MemberCollectiveId, sequelizeParams);
+    const memberCollectiveUser = await models.User.findByPk(member.CreatedByUserId, sequelizeParams);
```

Comment:
`CreatedByUserId` fields should not be used for such thing, we should query `User` with the `CollectiveId` instead.

Reply:
like this `await models.User.findByPk(member.CollectiveId, sequelizeParams)` ?

Reply:
`member.CollectiveId` is a collective ID (not a user ID). You can try something like this instead:

```es6
let memberCollectiveUser;
if (memberCollective.type === types.USER && !memberCollective.isIncognito) {
  memberCollectiveUser = await models.User.findOne({ where: { CollectiveId: memberCollective.id } });
}
```

Reply:
Thanks